qa/L0_response_cache/test.sh

```diff
@@ -140,52 +170,90 @@ function check_server_expected_failure {
     fi
 }
 
-MODEL_DIR="${PWD}/models"
-mkdir -p "${MODEL_DIR}/decoupled_cache/1"
-mkdir -p "${MODEL_DIR}/identity_cache/1"
-
-# Check that server fails to start for a "decoupled" model with cache enabled
-EXTRA_ARGS="--model-control-mode=explicit --load-model=decoupled_cache"
-
-SERVER=/opt/tritonserver/bin/tritonserver
-SERVER_ARGS="--model-repository=${MODEL_DIR} --response-cache-byte-size=8192 ${EXTRA_ARGS}"
-SERVER_LOG="./inference_server.log"
-source ../common/util.sh
-run_server
-if [ "$SERVER_PID" != "0" ]; then
-    echo -e "\n***\n*** Failed: $SERVER started successfully when it was expected to fail\n***"
-    cat $SERVER_LOG
+function check_server_failure_decoupled_model {
+  MODEL_REPOSITORY="${1}"
+  MODEL_CONTROL_MODE="${2}"
+  MODEL_LOAD_TYPE="${3}"
+  EXTRA_ARGS="--model-control-mode=${MODEL_CONTROL_MODE} --load-model=${MODEL_LOAD_TYPE}"
+  SERVER_ARGS="--model-repository=${MODEL_REPOSITORY} --cache-config local,size=1048576000 ${EXTRA_ARGS}"
+  source ../common/util.sh
```

Comment:
Why is this sourced as part of the function? This should be moved higher up and outside of the function.

Reply:
Actually looks like it's already included higher up, so this can just be removed.

Reply:
I removed the SERVER_ARGS defined outside the function as it is not used. This should be inside the function as ${MODEL_REPOSITORY} changes for decoupled_cache test case and ensemble cache testcase. Other testcases use different SERVER_ARGS.